### PR TITLE
WIP refactor: renames all keycloak providers/tokens/services.

### PR DESCRIPTION
**Pull request by hemanik**
This PR renames all references to Keycloak providers/tokens/services to make it more generic and move towards decoupling with Keycloak for the login process. 

Fixes: #583

### CODE REVIEW

File: controller/users.go [hunk 1]
```diff
@@ -925,9 +925,9 @@ func (c *UsersController) VerifyEmail(ctx *app.VerifyEmailUsersContext) error {
 			// try hitting the admin user endpoint only if getting a PAT
 			// was successful.
 
-			usersEndpoint, err := c.config.GetKeycloakEndpointUsers(ctx.RequestData)
+			usersEndpoint, err := c.config.GetOAuthServiceEndpointUsers(ctx.RequestData)
```
Comment (golangci) [inline controller/users.go]: ineffectual assignment to `err`

File: login/service.go [hunk 2]
```diff
@@ -420,30 +420,30 @@ func (keycloak *KeycloakOAuthProvider) updateUserInKeycloak(ctx context.Context,
 		// try hitting the admin user endpoint only if getting a PAT
 		// was successful.
 
-		usersEndpoint, err := config.GetKeycloakEndpointUsers(request)
+		usersEndpoint, err := config.GetOAuthServiceEndpointUsers(request)
```
Comment (golangci) [inline login/service.go]: ineffectual assignment to `err`

File: login/profile_blackbox_test.go [hunk 3]
```diff
@@ -92,13 +92,13 @@ func (s *ProfileBlackBoxTest) generateAccessToken() (*string, error) {
 	r := &goa.RequestData{
 		Request: &http.Request{Host: "api.example.org"},
 	}
-	tokenEndpoint, err := s.Config.GetKeycloakEndpointToken(r)
+	tokenEndpoint, err := s.Config.GetOAuthServiceEndpointToken(r)
```
Comment (golangci) [inline login/profile_blackbox_test.go]: ineffectual assignment to `err`

File: login/profile_user_blackbox_test.go [hunk 4]
```diff
@@ -123,16 +123,16 @@ func (s *ProfileUserBlackBoxTest) TestKeycloakAddUser() {
...
 	//"https://sso.prod-preview.openshift.io/auth/admin/realms/fabric8/users/" + identityID + "/federated-identity/rhd"
-	linkURL, err := s.Config.GetKeycloakEndpointLinkIDP(r, identityID, idpName)
+	linkURL, err := s.Config.GetOAuthServiceEndpointLinkIDP(r, identityID, idpName)
 	require.Nil(s.T(), err)
 
 	err = s.idpLinkService.Create(context.Background(), &linkRequest, s.protectedAccessToken, linkURL)
 	require.Nil(s.T(), err)
 
-	userURL = s.updateExistingUser(&testKeycloakUserData)
+	userURL = s.updateExistingUser(&testOAuthServiceUserData)
```
Comment (golangci) [inline login/profile_user_blackbox_test.go]: ineffectual assignment to `userURL`

File: controller/users.go [hunk 5]
```diff
@@ -925,9 +925,11 @@ func (c *UsersController) VerifyEmail(ctx *app.VerifyEmailUsersContext) error {
 			// try hitting the admin user endpoint only if getting a PAT
 			// was successful.
 
-			usersEndpoint, err := c.config.GetKeycloakEndpointUsers(ctx.RequestData)
+			if usersEndpoint, err := c.config.GetOAuthServiceEndpointUsers(ctx.RequestData); err != nil {
```
Comment (golangci) [inline controller/users.go]: ineffectual assignment to `usersEndpoint`

File: account/repository/identity.go [hunk 6]
```diff
@@ -60,17 +60,17 @@ func (u NullUUID) Value() (driver.Value, error) {
...
 	gormsupport.Lifecycle
-	// This is the ID PK field. For identities provided by Keycloak this ID equals to the Keycloak. For other types of IDP (github, oso, etc) this ID is generated automaticaly
+	// This is the ID PK field. For types of IDP (github, oso, etc) this ID is generated automaticaly
 	ID uuid.UUID `sql:"type:uuid default uuid_generate_v4()" gorm:"primary_key"`
 	// The username of the Identity
 	Username string
 	// Whether username has been updated.
 	RegistrationCompleted bool `gorm:"column:registration_completed"`
-	// ProviderType The type of provider, such as "keycloak", "github", "oso", etc
+	// ProviderType The type of provider, such as "github", "oso", etc
```
Comment (alexeykazakov) [inline account/repository/identity.go]: I would mention that "kc" type is used for the main Identity Provider.

File: login/link/keycloak_idp.go [hunk 7]
```diff
@@ -16,38 +16,38 @@ import (
 {"identityProvider":"rhd","userId":"<USER_ID>","userName":"<USERNAME>"}
 */
 
-// KeycloakLinkIDPRequest represents standard Keycloak User profile api request payload
-type KeycloakLinkIDPRequest struct {
+// OAuthServiceLinkIDPRequest represents standard OAuth service User profile api request payload
+type OAuthServiceLinkIDPRequest struct {
```
Comment (alexeykazakov) [inline login/link/keycloak_idp.go]: keycloak_idp.go is for Keycloak specific code. It works only with Keycloak. I would not rename it.

File: migration/sql-files/003-space-resources.sql [hunk 8]
```diff
@@ -1,4 +1,4 @@
--- Create space resource table for Keycloak resources associated with spaces
+-- Create space resource table for oauth service resources associated with spaces
```
Comment (alexeykazakov) [inline migration/sql-files/003-space-resources.sql]: That was Keycloak specific table. Keep Keycloak reference here.

**Comment by fabric8cd** [comment]
@hemanik snapshot fabric8-auth image is available for testing.  `docker pull docker.io/fabric8/fabric8-auth:SNAPSHOT-PR-594-5`

**Comment by xcoulon** [comment]
[test]

**Comment by fabric8cd** [comment]
@hemanik snapshot fabric8-auth image is available for testing.  `docker pull docker.io/fabric8/fabric8-auth:SNAPSHOT-PR-594-6`

File: openshift/auth.app.yaml [hunk 9]
```diff
@@ -49,16 +49,16 @@ objects:
...
                 name: auth
                 key: db.password
-          - name: AUTH_KEYCLOAK_REALM
+          - name: AUTH_OAUTH_SERVICE_REALM
             valueFrom:
               secretKeyRef:
                 name: auth
-                key: keycloak.realm
-          - name: AUTH_KEYCLOAK_SECRET
+                key: oauth.service.realm
```
Comment (xcoulon) [inline openshift/auth.app.yaml]: if we change the secret keys here, then we'll need to change the secrets that are deployed on prod and prod-preview.

File: test/login.go [hunk 10]
```diff
@@ -116,6 +116,6 @@ type DummyOSORegistrationApp struct {
 	Err    error
 }
 
-func (regApp *DummyOSORegistrationApp) LoadOSOSubscriptionStatus(ctx context.Context, config login.Configuration, keycloakToken oauth2.Token) (string, error) {
+func (regApp *DummyOSORegistrationApp) LoadOSOSubscriptionStatus(ctx context.Context, config login.Configuration, OAuth2Token oauth2.Token) (string, error) {
```
Comment (xcoulon) [inline test/login.go]: this variable's name should start with a lowercase (ie, `oauth2Token`), or even just `t`, as short names are usually preferred

File: token/token.go [hunk 11]
```diff
@@ -480,12 +480,12 @@ func (mgm *tokenManager) GenerateUserTokenForIdentity(ctx context.Context, ident
 	return token, nil
 }
 
-// GenerateUnsignedUserAccessToken generates an unsigned OAuth2 user access token for the given identity based on the Keycloak token
-func (mgm *tokenManager) GenerateUnsignedUserAccessToken(ctx context.Context, keycloakAccessToken string, identity *repository.Identity) (*jwt.Token, error) {
+// GenerateUnsignedUserAccessToken generates an unsigned OAuth2 user access token for the given identity based on the OAuth2 token
+func (mgm *tokenManager) GenerateUnsignedUserAccessToken(ctx context.Context, OAuth2AccessToken string, identity *repository.Identity) (*jwt.Token, error) {
```
Comment (xcoulon) [inline token/token.go]: here, `OAuth2AccessToken` should also start with a lowercase, too, eg: `accessToken`

File: token/token.go [hunk 12]
```diff
@@ -600,12 +600,12 @@ func (mgm *tokenManager) GenerateUnsignedUserAccessTokenForIdentity(ctx context.
 	return token, nil
 }
 
-// GenerateUnsignedUserRefreshToken generates an unsigned OAuth2 user refresh token for the given identity based on the Keycloak token
-func (mgm *tokenManager) GenerateUnsignedUserRefreshToken(ctx context.Context, keycloakRefreshToken string, identity *repository.Identity) (*jwt.Token, error) {
+// GenerateUnsignedUserRefreshToken generates an unsigned OAuth2 user refresh token for the given identity based on the OAuth2 token
+func (mgm *tokenManager) GenerateUnsignedUserRefreshToken(ctx context.Context, OAuth2RefreshToken string, identity *repository.Identity) (*jwt.Token, error) {
```
Comment (xcoulon) [inline token/token.go]: same comment here: `refreshToken`

File: account/repository/identity.go [hunk 13]
```diff
@@ -22,8 +22,8 @@ import (
 )
 
 const (
-	// KeycloakIDP is the name of the main Keycloak Identity Provider
-	KeycloakIDP string = "kc"
+	// OAuthServiceIDP is the name of the main OAuthService Identity Provider
+	OAuthServiceIDP string = "kc"
```
Comment (xcoulon) [inline account/repository/identity.go]: I find the name `OAuthServiceIDP` a bit confusing as it mixes a protocol with a service ("identity provider"). May I suggest that we name it to something like `DefaultIDP` instead ? or maybe `OSIOIdentityProvider`, as a reference to the main/internal IDP compared to external IDPs such as GitHub, etc. ? 
(this also joins @alexeykazakov's comment in https://github.com/fabric8-services/fabric8-auth/pull/594/files#r209036273)

**Comment by fabric8cd** [comment]
@hemanik snapshot fabric8-auth image is available for testing.  `docker pull docker.io/fabric8/fabric8-auth:SNAPSHOT-PR-594-7`

File: configuration/configuration.go [hunk 14]
```diff
@@ -72,23 +72,23 @@ const (
...
+	varOAuthServiceTesUser2Name     = "oauth.service.testuser2.name"
+	varOAuthServiceTesUser2Secret   = "oauth.service.testuser2.secret"
+	varOAuthServiceURL              = "oauth.service.url"
+	varOAuthServiceEndpointAdmin    = "oauth.service.endpoint.admin"
+	varOAuthServiceEndpointAuth     = "oauth.service.endpoint.auth"
+	varOAuthServiceEndpointToken    = "oauth.service.endpoint.token"
+	varOAuthServiceEndpointUserinfo = "oauth.service.endpoint.userinfo"
+	varOAuthServiceEndpointBroker   = "oauth.service.endpoint.broker"
+	varOAuthServiceEndpointAccount  = "oauth.service.endpoint.account"
+	varOAuthServiceEndpointLogout   = "oauth.service.endpoint.logout"
```
Comment (sbose78) [inline configuration/configuration.go]: Fine to change the variable names, not the values yet, please.

File: config.yaml [hunk 15]
```diff
@@ -93,17 +93,17 @@ serviceaccount.privatekeyid: 9MLnViaRkhVj1GT9kpWUkwHIwUD-wZfUxR-3CpkE-Xs
...
-keycloak.testuser2.name : testuser2
-keycloak.testuser2.secret : testuser2
+oauth.service.client.id : fabric8-online-platform
+oauth.service.secret : 7a3d5a00-7f80-40cf-8781-b5b6f2dfd1bd
+oauth.service.domain.prefix : sso
+oauth.service.realm : fabric8
+oauth.service.testuser.name : testuser
+oauth.service.testuser.secret : testuser
+oauth.service.testuser2.name : testuser2
+oauth.service.testuser2.secret : testuser2
```
Comment (sbose78) [inline config.yaml]: Let;s not change these yet.

File: configuration/configuration.go (hunk 14 above)
Comment (sbose78) [reply configuration/configuration.go]: The migration path should be : 
- add new env variables in our cluster, while the old ones still exist.
- remove refs to old env variables in our code.

**Comment by fabric8cd** [comment]
@hemanik snapshot fabric8-auth image is available for testing.  `docker pull docker.io/fabric8/fabric8-auth:SNAPSHOT-PR-594-8`

**Comment by fabric8cd** [comment]
@hemanik snapshot fabric8-auth image is available for testing.  `docker pull docker.io/fabric8/fabric8-auth:SNAPSHOT-PR-594-9`

**Comment by hemanik** [comment]
[test]

**Comment by sbose78** [comment]
Putting this on hold till I figure out changes needed in here https://github.com/fabric8-services/fabric8-auth/pull/618

**Comment by sbose78** [comment]
Changes from here are being taken up in other PRs.

**Pull request closed without merging**